app/forms/openid_connect_authorize_form.rb

```diff
@@ -56,6 +56,13 @@ def submit(user, rails_session_id)
     FormResponse.new(success: success, errors: errors.messages, extra: extra_analytics_attributes)
   end
 
+  # Similar to submit, but only helps generate errors
+  def check_submit
```

Comment:
Rename to `validate`? Also, since we are now validating earlier in the process, what do you think about simplifying `#submit` to this:
```ruby
link_identity_to_client_id(user, rails_session_id)
FormResponse.new(success: true, errors: {}, extra: extra_analytics_attributes)
```